### ISSUE TITLE
Annotations appear twice in Stackdriver Trace

### ISSUE THREAD

**Issue by Sebruck** [census-instrumentation/opencensus-java#1201]
Please answer these questions before submitting a bug report.

### What version of OpenCensus are you using?
0.13.2

### What JVM are you using (`java -version`)?
8

### What did you do?
Simply starting a span and adding an annotation via `span.addAnnotation("start")`. Exporting it via the Stackdriver exporter.

### What did you expect to see?
The annotation appears once in the Stackdriver Trace UI

### What did you see instead?
The annotation is *sometimes* duplicated with the exact same timestamp (sometimes it's only once there). When I add the Logging Exporter, I can see that the annotation is only once attached to the span.

<img width="338" alt="screen shot 2018-05-17 at 14 45 08" src="https://user-images.githubusercontent.com/2050823/40178343-873b67f6-59e1-11e8-9adb-567d4f239278.png">

**Comment by mtwo** [comment]
Huh, that's no good. We just filed a bug to track this internally and will take a look ASAP.

**Comment by Sebruck** [comment]
Thanks for having a look!

**Comment by askingcat** [comment]
I can confirm this behaviour as well. I have been running a Jaeger instance (using the Zipkin exporter) at the same time as well and the duplicated annotations only showed up in the Stackdriver UI.

**Comment by bogdandrutu** [comment]
Hi all,

I think we found the issue. We believe this is because of failures/retries (one deadline request which gets to the backend and is executed) then we retry to send the same Spans. In the backend SD does a proto merge in this case which for repeated fields like annotation this cause duplicates.

SD team is working to a backend fix to avoid this problem and I started a discussion on the specs repo about OC supporting retries (https://github.com/census-instrumentation/opencensus-specs/issues/119)
